lib/src/jpegr.cpp

```diff
@@ -494,6 +495,11 @@ uhdr_error_info_t JpegR::convertYuv(uhdr_raw_image_t* image, uhdr_color_gamut_t
 
 uhdr_error_info_t JpegR::compressGainMap(uhdr_raw_image_t* gainmap_img,
                                          JpegEncoderHelper* jpeg_enc_obj) {
+  if (!kWriteXmpMetadata) {
+    std::shared_ptr<DataStruct> icc = IccHelper::writeIccProfile(gainmap_img->ct, gainmap_img->cg);
```

Comment:
Thank you!

There is another flag in the iso metadata `useBaseColorSpace`. The gain map / icc logic should also be hidden when that flag is false, and the metadata encode / decode part also needs modification to support this feature.

Reply:
useBaseColorSpace is not configurable on the encoding side. as hdr intent is being converted to sdr intent and sdr image is base image, gain map application space shall always be base color space. hence useBaseColorSpace is always true on encoding side.
The change associated with selecting useBaseColorSpace dynamically are in the follow up CL. This change just signals hdr intent color space in gainmap image icc. In the next CL as useBaseColorSpace is selected dynamically and the encode / decode part is also modified. Further this dynamic selection is disabled if kWriteXmpMetadata is enabled.

Reply:
OK

Reply:
Dichen, this pr, #336, perhaps addresses some of the concerns you raised.